server/service.go

```diff
@@ -109,8 +109,13 @@ func (m *BoostService) RegisterValidatorV1(ctx context.Context, message types.Re
 		return nil, rpcErrInvalidSignature
 	}
 
-	ok := false // at least one builder has returned true
-	var lastRelayError error
+	type safeRegisterContainer struct {
+		mu             sync.Mutex
+		ok             bool // at least one builder has returned true
+		lastRelayError error
+	}
+	container := safeRegisterContainer{}
```

Comment:
Why did you choose to create the container vs just adding the mutex?

Reply:
As the `ok` and `lastRelayError` fields both need to be protected by the mutex, I assumed it made more sense to combine them in a structure.
It shows that this mutex will "only" be used to protect these variables only.

Let me know if you disagree with this !

Reply:
Makes sense :)